src/Data/Nat/Properties.agda

```diff
@@ -1598,11 +1596,21 @@ m≤n⇒n∸m≤n (s≤s m≤n) = m≤n⇒m≤1+n (m≤n⇒n∸m≤n m≤n)
   (m + n) ∸ o          ≡⟨ +-∸-assoc m o≤n ⟩
   m + (n ∸ o)          ∎
 
+m≤n+o⇒m∸n≤o : ∀ m n {o} → m ≤ n + o → m ∸ n ≤ o
```

Comment:
We are really inconsistent with which arguments are explicit/implicit in this and the three proofs immediately below...

Reply:
Yes, some if that us historical. Some if it may be carelessness on my part, some if it may be because what's inferrable does actually vary ... will investigate.

Reply:
So there are (at least) two instances where it seems useful to have the first (left-hand) arguments explicit, namely to each of `_+_` and `_*_`, because Agda doesn't always seem able to infer those, even if they are subsequently governed by  a predicate... the 'usual' function inversion problem. And as elsewhere in the library, custom seems to dictate using explicit arguments when doing an explicit pattern match. That said, this is an example where the witness to the the relation could drive the proof (it's not irrelevant/erasable), so maybe a better version is possible.

Reply:
I think I have a tidying up for this now, but the still there seems to be some variation.

Reply:
Okay great thanks!